### PR TITLE
Unquarantine tests / skip mslocaldb helix tests

### LINKED ISSUE #31729: RazorPagesTemplate_IdentityWeb_BuildsAndPublishes template test flakiness
```
[xUnit.net 00:00:15.84]     Templates.Test.RazorPagesTemplateTest.RazorPagesTemplate_IdentityWeb_BuildsAndPublishes(auth: "SingleOrg", args: ["--calls-graph"]) [FAIL]
[xUnit.net 00:00:15.84]       Process exited with code 134
[xUnit.net 00:00:15.84]       StdErr: malloc_consolidate(): invalid chunk size
[xUnit.net 00:00:15.84]       
[xUnit.net 00:00:15.84]       StdOut: Template Name                         Short Name      Language    Tags          
[xUnit.net 00:00:15.84]       ------------------------------------  --------------  ----------  --------------
[xUnit.net 00:00:15.84]       Console Application                   console         [C#],F#,VB  Common/Console
[xUnit.net 00:00:15.84]       Class library                         classlib        [C#],F#,VB  Common/Library
[xUnit.net 00:00:15.84]       MSTest Test Project                   mstest          [C#],F#,VB  Test/MSTest   
[xUnit.net 00:00:15.84]       NUnit 3 Test Item                     nunit-test      [C#],F#,VB  Test/NUnit    
[xUnit.net 00:00:15.84]       NUnit 3 Test Project                  nunit           [C#],F#,VB  Test/NUnit    
[xUnit.net 00:00:15.84]       xUnit Test Project                    xunit           [C#],F#,VB  Test/xUnit    
[xUnit.net 00:00:15.84]       Razor Component                       razorcomponent  [C#]        Web/ASP.NET   
[xUnit.net 00:00:15.84]       Razor Page                            page            [C#]        Web/ASP.NET   
[xUnit.net 00:00:15.84]       MVC ViewImports                       viewimports     [C#]        Web/ASP.NET   
[xUnit.net 00:00:15.84]       MVC ViewStart                         viewstart       [C#]        Web/ASP.NET   
[xUnit.net 00:00:15.84]       ASP.NET Core with Angular             angular         [C#]        Web/MVC/SPA   
[xUnit.net 00:00:15.84]       ASP.NET Core with React.js            react           [C#]        Web/MVC/SPA   
[xUnit.net 00:00:15.84]       ASP.NET Core with React.js and Redux  reactredux      [C#]        Web/MVC/SPA   
[xUnit.net 00:00:15.84]       TypeScript file                       tsfile                      Web/Script    
[xUnit.net 00:00:15.84]       LESS Style Sheet                      less                        Web/Style     
[xUnit.net 00:00:15.84]       SCSS Style Sheet (SASS)               scss                        Web/Style     
[xUnit.net 00:00:15.84]       Protocol Buffer File                  proto                       Web/gRPC      
[xUnit.net 00:00:15.84]       
[xUnit.net 00:00:15.84]       Examples:
[xUnit.net 00:00:15.84]           dotnet new nunit 
[xUnit.net 00:00:15.84]           dotnet new react --auth Individual
[xUnit.net 00:00:15.84]           dotnet new --help
[xUnit.net 00:00:15.85]           dotnet new proto --help
[xUnit.net 00:00:15.85]       
[xUnit.net 00:00:15.85]       Expected: True
[xUnit.net 00:00:15.85]       Actual:   False
[xUnit.net 00:00:15.85]       Stack Trace:
[xUnit.net 00:00:15.85]         /_/src/ProjectTemplates/Shared/TemplatePackageInstaller.cs(153,0): at Templates.Test.Helpers.TemplatePackageInstaller.InstallTemplatePackages(ITestOutputHelper output)
[xUnit.net 00:00:15.85]         /_/src/ProjectTemplates/Shared/TemplatePackageInstaller.cs(64,0): at Templates.Test.Helpers.TemplatePackageInstaller.EnsureTemplatingEngineInitializedAsync(ITestOutputHelper output)
[xUnit.net 00:00:15.85]         /_/src/ProjectTemplates/Shared/ProjectFactoryFixture.cs(28,0): at Templates.Test.Helpers.ProjectFactoryFixture.GetOrCreateProject(String projectKey, ITestOutputHelper output)
[xUnit.net 00:00:15.85]         /_/src/ProjectTemplates/test/RazorPagesTemplateTest.cs(250,0): at Templates.Test.RazorPagesTemplateTest.BuildAndPublishRazorPagesTemplate(String auth, String[] args)
[xUnit.net 00:00:15.85]         --- End of stack trace from previous location ---
  Failed Templates.Test.RazorPagesTemplateTest.RazorPagesTemplate_IdentityWeb_BuildsAndPublishes(auth: "SingleOrg", args: ["--calls-graph"]) [14 s]
  Error Message:
   Process exited with code 134
StdErr: malloc_consolidate(): invalid chunk size
```

https://dev.azure.com/dnceng/public/_build/results?buildId=1083545&view=logs&j=3f6d4e0f-1b71-56b5-361e-d95b6e6da15a&t=d5bf30bc-9e5a-596c-d1c8-39e2dfa42d49

### LINKED ISSUE #34284: FileLoggerProcessorTests.RollsTextFiles
## Failing Test(s)

Microsoft.AspNetCore.HttpLogging.FileLoggerProcessorTests.RollsTextFiles

## Error Message
```text
   Assert.Equal() Failure
          â†“ (pos 0)
Expected: Message two\n
Actual:   
          â†‘ (pos 0)
  Stack Trace:
     at Microsoft.AspNetCore.HttpLogging.FileLoggerProcessorTests.RollsTextFiles() in /_/src/Middleware/HttpLogging/test/FileLoggerProcessorTests.cs:line 117
```
https://github.com/dotnet/aspnetcore/blob/69eb4b38bca90bb830f290a6eaa5a61650dc87ac/src/Middleware/HttpLogging/test/FileLoggerProcessorTests.cs#L117
## Build

https://dev.azure.com/dnceng/public/_build/results?buildId=1231336&view=ms.vss-test-web.build-test-results-tab&runId=36670362&resultId=110218&paneView=debug
(Alpine.312.Amd64.Open)Ubuntu.1804.Amd64.Open

### LINKED ISSUE #37930: Http2ConnectionTests.ResponseTrailers_MultipleStreams_Reset test failure
I'm guessing https://github.com/dotnet/aspnetcore/pull/37627 regressed this test as it modified behavior in this area and was merged recently.

## Failing Test(s)

Microsoft.AspNetCore.Server.Kestrel.Core.Tests.Http2ConnectionTests.ResponseTrailers_MultipleStreams_Reset

## Error Message

```text
Assert.Equal() Failure
Expected: 1
Actual:   0
```

## Stacktrace

<details>

```text
   at Microsoft.AspNetCore.Server.Kestrel.Core.Tests.Http2ConnectionTests.ResponseTrailers_MultipleStreams_Reset() in /_/src/Servers/Kestrel/test/InMemory.FunctionalTests/Http2/Http2ConnectionTests.cs:line 301
--- End of stack trace from previous location ---
```

</details>

## Logs

<details>

```text
[0.000s] [TestLifetime] [Information] Starting test ResponseTrailers_MultipleStreams_Reset at 2021-10-29T13:01:32
[0.017s] [Microsoft.AspNetCore.Server.Kestrel.Http2] [Verbose] Connection id "TestConnectionId" sending SETTINGS frame for stream ID 0 with length 18 and flags NONE.
[0.017s] [Microsoft.AspNetCore.Server.Kestrel.Http2] [Verbose] Connection id "TestConnectionId" sending WINDOW_UPDATE frame for stream ID 0 with length 4 and flags 0x0.
[0.017s] [Microsoft.AspNetCore.Server.Kestrel.Http2] [Verbose] Connection id "TestConnectionId" received SETTINGS frame for stream ID 0 with length 0 and flags NONE.
[0.017s] [Microsoft.AspNetCore.Server.Kestrel.Http2] [Verbose] Connection id "TestConnectionId" sending SETTINGS frame for stream ID 0 with length 0 and flags ACK.
[0.017s] [Microsoft.AspNetCore.Server.Kestrel.Http2] [Verbose] Connection id "TestConnectionId" received HEADERS frame for stream ID 1 with length 97 and flags END_STREAM, END_HEADERS.
[0.020s] [Microsoft.AspNetCore.Server.Kestrel.Http2] [Verbose] Connection id "TestConnectionId" sending HEADERS frame for stream ID 1 with length 36 and flags END_HEADERS.
[0.020s] [Microsoft.AspNetCore.Server.Kestrel.Http2] [Verbose] Connection id "TestConnectionId" sending HEADERS frame for stream ID 1 with length 16 and flags END_STREAM, END_HEADERS.
[0.098s] [Microsoft.AspNetCore.Server.Kestrel.Http2] [Verbose] Connection id "TestConnectionId" received PING frame for stream ID 0 with length 8 and flags NONE.
[0.099s] [Microsoft.AspNetCore.Server.Kestrel.Http2] [Verbose] Connection id "TestConnectionId" sending PING frame for stream ID 0 with length 8 and flags ACK.
[0.099s] [Microsoft.AspNetCore.Server.Kestrel.Http2] [Verbose] Connection id "TestConnectionId" received PING frame for stream ID 0 with length 8 and flags NONE.
[0.099s] [Microsoft.AspNetCore.Server.Kestrel.Http2] [Verbose] Connection id "TestConnectionId" sending PING frame for stream ID 0 with length 8 and flags ACK.
[0.099s] [Microsoft.AspNetCore.Server.Kestrel.Http2] [Verbose] Connection id "TestConnectionId" received HEADERS frame for stream ID 3 with length 5 and flags END_STREAM, END_HEADERS.
[0.100s] [Microsoft.AspNetCore.Server.Kestrel.Http2] [Verbose] Connection id "TestConnectionId" sending HEADERS frame for stream ID 3 with length 6 and flags END_HEADERS.
[0.100s] [Microsoft.AspNetCore.Server.Kestrel.Http2] [Verbose] Connection id "TestConnectionId" sending HEADERS frame for stream ID 3 with length 16 and flags END_STREAM, END_HEADERS.
[0.100s] [Microsoft.AspNetCore.Server.Kestrel.Http2] [Verbose] Connection id "TestConnectionId" received PING frame for stream ID 0 with length 8 and flags NONE.
[0.100s] [Microsoft.AspNetCore.Server.Kestrel.Http2] [Verbose] Connection id "TestConnectionId" sending PING frame for stream ID 0 with length 8 and flags ACK.
[0.101s] [Microsoft.AspNetCore.Server.Kestrel.Http2] [Verbose] Connection id "TestConnectionId" received PING frame for stream ID 0 with length 8 and flags NONE.
[0.101s] [Microsoft.AspNetCore.Server.Kestrel.Http2] [Verbose] Connection id "TestConnectionId" sending PING frame for stream ID 0 with length 8 and flags ACK.
[0.170s] [Microsoft.AspNetCore.Server.Kestrel.Core.Tests.Http2ConnectionTests] [Error] Test threw an exception.
Xunit.Sdk.EqualException: Assert.Equal() Failure
Expected: 1
Actual:   0
   at Xunit.Assert.Equal[T](T expected, T actual, IEqualityComparer`1 comparer) in C:\Dev\xunit\xunit\src\xunit.assert\Asserts\EqualityAsserts.cs:line 40
   at Microsoft.AspNetCore.Server.Kestrel.Core.Tests.Http2ConnectionTests.ResponseTrailers_MultipleStreams_Reset() in /_/src/Servers/Kestrel/test/InMemory.FunctionalTests/Http2/Http2ConnectionTests.cs:line 301
   at Xunit.Sdk.TestInvoker`1.<>c__DisplayClass48_1.<<InvokeTestMethodAsync>b__1>d.MoveNext() in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\Runners\TestInvoker.cs:line 264
--- End of stack trace from previous location ---
   at Xunit.Sdk.ExecutionTimer.AggregateAsync(Func`1 asyncAction) in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\ExecutionTimer.cs:line 48
   at Xunit.Sdk.ExceptionAggregator.RunAsync(Func`1 code) in C:\Dev\xunit\xunit\src\xunit.core\Sdk\ExceptionAggregator.cs:line 90
[0.171s] [Microsoft.AspNetCore.Server.Kestrel.Http2] [Debug] Connection id "TestConnectionId" is closed. The last processed stream ID was 3.
[0.171s] [Microsoft.AspNetCore.Server.Kestrel.Http2] [Verbose] Connection id "TestConnectionId" sending GOAWAY frame for stream ID 0 with length 8 and flags 0x0.
[0.180s] [TestLifetime] [Information] Finished test ResponseTrailers_MultipleStreams_Reset in 0.1796598s
```
</details>

## Build

https://dev.azure.com/dnceng/public/_build/results?buildId=1446040&view=ms.vss-test-web.build-test-results-tab&runId=41665248&resultId=106843&paneView=debug

### PR DESCRIPTION
**Pull request by HaoK**
Fixes https://github.com/dotnet/aspnetcore/issues/37930
Fixes https://github.com/dotnet/aspnetcore/issues/34284
Fixes https://github.com/dotnet/aspnetcore/issues/31729
Fixes https://github.com/dotnet/aspnetcore/issues/37930

Parole for:
- 2 non playwright blazor template tests that were grouped in the playwright template test issue, these are fairly reliable and have been clean for 30 days
- FileLoggerProcessor test was improved by community PRs
- Http2ConnectionTest that was missing a test-fixed label that has been 30 days clean
- Switch to retry instead of perma quarantine for MaxRequestBufferSizeTests.LargeUpload as its been 30 days clean

image

### CODE REVIEW

**Review by halter73 — APPROVED** [review]

File: src/Servers/Kestrel/test/FunctionalTests/MaxRequestBufferSizeTests.cs [hunk 1]
```diff
@@ -115,7 +115,7 @@ public static IEnumerable<object[]> LargeUploadData
     }
     [Theory]
     [MemberData(nameof(LargeUploadData))]
-    [QuarantinedTest("This is inherently flaky and should never be unquarantined.")]
+    // This is inherently flaky and is relying on helix retry to pass consistently
```
Comment (halter73) [inline src/Servers/Kestrel/test/FunctionalTests/MaxRequestBufferSizeTests.cs:118]: I hope this works!

Comment (HaoK) [reply src/Servers/Kestrel/test/FunctionalTests/MaxRequestBufferSizeTests.cs:118]: So far so good, haven't had any of the retry tests get quarantined yet (knock on wood)

**Review by wtgodbe — APPROVED** [review]

**Comment by HaoK** [comment]
@dotnet/aspnet-blazor-eng fyi we've been seeing the playwright versions of BlazorTemplate tests fail recently, so I'm going to take it out of helix runs for now and file an issue if you want to bring these back in the future, I believe all of the tests in the playwright version of BlazorTemplates are quarantined anyways, so there should not be any loss of coverage

**Comment by HaoK** [comment]
Filed https://github.com/dotnet/aspnetcore/issues/38818 for tracking the playwright blazortemplate.tests being skipped on helix

**Comment by HaoK** [comment]
Looks like the hang issue is TestDependsOnMssql as Diagnostics functionals is also hanging, will skip and file an issue for that as well

**Comment by pranavkm** [comment]
> @dougbu backporting to release/6.0 failed, the patch most likely resulted in conflicts:

Some dolt updated this repo to use file scoped namespaces in main, so it's super hard to port these changes without conflicts. https://github.com/dotnet/aspnetcore/pull/40101 redoes this PR manually.

**Pull request merged**